src/ripple/ledger/CashDiff.h

```diff
@@ -0,0 +1,131 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_LEDGER_CASHDIFF_H_INCLUDED
+#define RIPPLE_LEDGER_CASHDIFF_H_INCLUDED
+
+#include <ripple/protocol/STAmount.h>
+#include <memory>                       // std::unique_ptr
+
+namespace ripple {
+
+class ReadView;
+
+namespace detail {
+
+class ApplyStateTable;
+
+}
+
+// Used by CashDiff to specify filters applied while processing differences.
+// Entries are bit flags that can be ANDed and ORed.
+enum class CashFilter : std::uint8_t
```

Comment:
CashDiff performs a similar function for offer crossing that FlowDebugInfo performs for `flow`: https://github.com/ripple/rippled/blob/develop/src/ripple/app/paths/impl/FlowDebugInfo.h#L41.  The primary difference being that CashDiff looks at changes to offers in addition to changes to trust lines and changes to the XRP balance in account roots.  Scott and I talked about whether I should modify his differencing tool.  We agreed that it would probably be better if I made my own.